pkg/storegateway/series_chunks.go

```diff
@@ -345,7 +355,7 @@ func (c *loadingSeriesChunksSetIterator) Next() (retHasNext bool) {
 	}
 
 	// Create a batched memory pool that can be released all at once.
-	chunksPool := &pool.BatchBytes{Delegate: c.chunksPool}
+	chunksPool := pool.NewSafeSlabPool[byte](chunkBytesSlicePool, chunkBytesSlabSize)
 
 	err := c.chunkReaders.load(nextSet.series, chunksPool, c.stats)
```

Comment:
I added a line to release the pool before returning in the error path but didn't find an easy way to test this because we're not injecting the pool wrapper. Maybe you had a way in mind that I missed?

Reply:
I think we already test almost all of this flow. There is `chunkReaderMock` which can return an error when its `load` is called. We can change it to also do some allocations before returning the error. Then the existing assertions for the balance of `chunkBytesSlicePool` should be sufficient. 

If you go this route it will be helpful to explain why we do an extra allocation in the error case of `chunkReaderMock.load`

Reply:
Actually, let me revert that line and follow up in a separate PR. Can't hurt to keep the scope as small as possible I think.